Cannot find name '$localize' when using vitest without explicit import

### Is this a bug report or a feature request?

- [x] Bug Report
- [ ] Feature Request

### Please provide the steps to reproduce the issue [Bug Report only]

1. Run ng new vitest-test 
2. Run ng add @angular/localize
3. Change signal in app.html to use $localize

### Please provide the expected behavior vs the actual behavior you encountered [Bug Report only]

I expected succesful run
Instead it only succeeds when i add `import '@angular/localize'` to app.ts, I can't do it using `setupFiles` option.
I suspect that it's because it cant pick up this automatic import when it builds and so i can do it only in app.ts, but i dont have to repeat this import for other components.
Adding vitest configuration with `globals` option enabled doesn't help either.

### Please provide a screenshot if possible [Bug Report only]

_No response_

### Please provide the exception or error you saw [Bug Report only]

```true
######/vitest-test>ng test
Automatically searching for and using Vitest configuration file.
Application bundle generation failed. [0.533 seconds] - 2026-01-06T19:48:08.764Z

✘ [ERROR] TS2304: Cannot find name '$localize'. [plugin angular-compiler]

    src/app/app.ts:11:36:
      11 │   protected readonly title = signal($localize`vitest-test`);
         ╵                                     ~~~~~~~~~


Watch mode enabled. Watching for file changes...
```

### Is this a browser-specific issue? If so, please specify the device, browser, and version. [Bug Report only]

```true
It's vitest related
```

### Description [Feature Request only]

_No response_

### Proposed solution [Feature Request only]

_No response_

### Alternatives considered [Feature Request only]

_No response_

--- a/packages/localize/schematics/ng-add/index.ts
+++ b/packages/localize/schematics/ng-add/index.ts
@@ -79,13 +79,26 @@ function addTypeScriptConfigTypes(projectName: string): Rule {
         case AngularBuilder.BrowserEsbuild:
         case AngularBuilder.Browser:
         case AngularBuilder.Application:
-        case AngularBuilder.BuildApplication:
+        case AngularBuilder.BuildKarma:
+        case AngularBuilder.BuildApplication: {
           const value = target.options?.['tsConfig'];
           if (typeof value === 'string') {
             tsConfigFiles.add(value);
           }
 
           break;
+        }
+        case AngularBuilder.BuildUnitTest: {
+          const value = target.options?.['tsConfig'];
+          if (typeof value === 'string') {
+            tsConfigFiles.add(value);
+          } else {
+            // Defaults to tsconfig in project root
+            tsConfigFiles.add((project.root || '.') + '/tsconfig.spec.json');
+          }
+
+          break;
+        }
       }
 
       if (

--- a/packages/localize/schematics/ng-add/index_spec.ts
+++ b/packages/localize/schematics/ng-add/index_spec.ts
@@ -73,16 +73,26 @@ describe('ng-add schematic', () => {
               test: {
                 builder: '@angular-devkit/build-angular:karma',
                 options: {
-                  tsConfig: './tsconfig.spec.json',
+                  tsConfig: './tsconfig.spec-k1.json',
                   polyfills: 'zone.js',
                 },
               },
               testKarmaBuild: {
                 builder: '@angular/build:karma',
                 options: {
-                  tsConfig: './tsconfig.spec.json',
+                  tsConfig: './tsconfig.spec-k2.json',
                   polyfills: 'zone.js',
                 },
+              },
+              testBuildUnitTestA: {
+                builder: '@angular/build:unit-test',
+                options: {
+                  tsConfig: './tsconfig.spec-a.json',
+                },
+              },
+              testBuildUnitTestB: {
+                builder: '@angular/build:unit-test',
+                options: {},
               },
               server: {
                 builder: '@angular-devkit/build-angular:server',
@@ -194,7 +204,55 @@ describe('ng-add schematic', () => {
     expect(polyfills).toEqual(['@angular/localize/init']);
   });
 
-  it(`should add '@angular/localize' in 'types' tsconfigs referenced in karma builder`, async () => {
+  it(`should add '@angular/localize' in 'types' tsconfigs referenced in devkit karma builder`, async () => {
+    const tsConfig = JSON.stringify({
+      compilerOptions: {
+        types: ['node'],
+      },
+    });
+
+    host.create('tsconfig.spec-k1.json', tsConfig);
+
+    host = await schematicRunner.runSchematic('ng-add', defaultOptions, host);
+    const {compilerOptions} = host.readJson('tsconfig.spec-k1.json') as TsConfig;
+    const types = compilerOptions?.types;
+    expect(types).toContain('@angular/localize');
+    expect(types).toHaveSize(2);
+  });
+
+  it(`should add '@angular/localize' in 'types' tsconfigs referenced in build karma builder`, async () => {
+    const tsConfig = JSON.stringify({
+      compilerOptions: {
+        types: ['node'],
+      },
+    });
+
+    host.create('tsconfig.spec-k2.json', tsConfig);
+
+    host = await schematicRunner.runSchematic('ng-add', defaultOptions, host);
+    const {compilerOptions} = host.readJson('tsconfig.spec-k2.json') as TsConfig;
+    const types = compilerOptions?.types;
+    expect(types).toContain('@angular/localize');
+    expect(types).toHaveSize(2);
+  });
+
+  it(`should add '@angular/localize' in 'types' tsconfigs referenced in unit-test builder`, async () => {
+    const tsConfig = JSON.stringify({
+      compilerOptions: {
+        types: ['node'],
+      },
+    });
+
+    host.create('tsconfig.spec-a.json', tsConfig);
+
+    host = await schematicRunner.runSchematic('ng-add', defaultOptions, host);
+    const {compilerOptions} = host.readJson('tsconfig.spec-a.json') as TsConfig;
+    const types = compilerOptions?.types;
+    expect(types).toContain('@angular/localize');
+    expect(types).toHaveSize(2);
+  });
+
+  it(`should add '@angular/localize' in 'types' tsconfigs default for unit-test builder`, async () => {
     const tsConfig = JSON.stringify({
       compilerOptions: {
         types: ['node'],

